Source/Orts.Simulation/Simulation/RollingStocks/MSTSSteamLocomotive.cs

```diff
@@ -5418,42 +5691,133 @@ public override void AdvancedAdhesion(float elapsedClockSeconds)
                 float AxleMomentInertia = (WheelWeightKG * AxleRadiusM * AxleRadiusM) / 2.0f;
                 float TotalWheelMomentofInertia = WheelMomentInertia + AxleMomentInertia; // Total MoI for generic wheelset
 
-                SteamDrvWheelWeightLbs = Kg.ToLb(DrvWheelWeightKg / LocoNumDrvAxles); // Calculate the weight per axle (used in MSTSLocomotive for friction calculatons)
+                SteamDrvWheelWeightLbs = Kg.ToLb(DrvWheelWeightKg / SteamEngines[numberofengine].NumberDriveWheelAxles); // Calculate the weight per axle (used in MSTSLocomotive for friction calculatons)
 
                 // The moment of inertia will be adjusted up or down compared to the size of the wheel on the player locomotive compared to the Generic wheel                
-                TotalWheelMomentofInertia *= DriverWheelRadiusM / WheelRadiusAssumptM;
+                TotalWheelMomentofInertia *= SteamEngines[numberofengine].DriveWheelRadiusM / WheelRadiusAssumptM;
 
                 // The moment of inertia needs to be increased by the number of wheel sets
-                TotalWheelMomentofInertia *= LocoNumDrvAxles;
+                TotalWheelMomentofInertia *= SteamEngines[numberofengine].NumberDriveWheelAxles;
 
                 // the inertia of the coupling rods can also be added
                 // Assume rods weigh approx 1500 lbs
                 // MoI = rod weight x stroke radius^2 (ie stroke / 2)
                 float RodWeightKG = Kg.FromLb(1500.0f);
                 // ???? For both compound and simple??????
-                float RodStrokeM = CylinderStrokeM / 2.0f;
-                float RodMomentInertia = RodWeightKG * RodStrokeM * RodStrokeM;
+                float RodStrokeM = SteamEngines[numberofengine].CylindersStrokeM / 2.0f;
+                float RodMomentInertia = 0;
+
+                if (SteamEngines[numberofengine].AuxiliarySteamEngineType != SteamEngine.AuxiliarySteamEngineTypes.Booster)
+                {
+                    RodMomentInertia = RodWeightKG * RodStrokeM * RodStrokeM;
+                }
 
                 float TotalMomentInertia = TotalWheelMomentofInertia + RodMomentInertia;
 
+                LocomotiveAxles[0].AxleWeightN = SteamEngines[numberofengine].AxleWeightN;
```

Comment:
This is wrong. You have to couple each SteamEngine to a different locomotive axle, and then do SteamEngines[numberofengine].Axle.AxleWeightN = ...